### PR TITLE
DDBTEAM-949: Changed ES to use index alias

### PR DESCRIPTION
**Pull request by cableman**
Added support for ES index alias in the populate service to enabled indexing without down time.

Requires: https://github.com/danskernesdigitalebibliotek/ddb-cover-service/pull/133

### CODE REVIEW

File: src/Service/PopulateService.php [hunk 1]
```diff
@@ -58,82 +62,92 @@ public function __construct(EntityManagerInterface $entityManager, SearchReposit
      *
      * @return void
      */
-    public function populate(string $index, int $record_id = -1): void
+    public function populate(int $record_id = -1): void
```
Comment (turegjorup) [inline src/Service/PopulateService.php:65]: If we plan to run this at intervals we should do a lock similar to vendor imports to ensure we don't run multiple populates in parallel.

File: src/Service/PopulateService.php [hunk 2]
```diff
@@ -58,82 +62,92 @@ public function __construct(EntityManagerInterface $entityManager, SearchReposit
...
-            $this->progressFinish();
-
-            return;
-        }
-
-        $entriesAdded = 0;
-        $currentId = 0;
+        $indexes = array_keys($this->indexManager->getAllIndexes());
+        foreach ($indexes as $indexName) {
+            // When aliases is configured this reset will create a new index in ES.
```
Comment (turegjorup) [inline src/Service/PopulateService.php:73]: 👍

**Review by turegjorup — CHANGES_REQUESTED** [review]

File: src/Service/PopulateService.php [hunk 3]
```diff
@@ -58,82 +62,92 @@ public function __construct(EntityManagerInterface $entityManager, SearchReposit
...
-
-            // Free up memory usages.
-            $this->entityManager->clear();
-            gc_collect_cycles();
+            $this->progressMessage(sprintf('<info>Refreshing</info> <comment>%s</comment>', $index->getName()));
+            $index->refresh();
 
-            $this->progressAdvance();
+            $this->progressMessage(sprintf('<info>Switching alias</info> <comment>%s -> %s</comment>', $index->getOriginalName(), $index->getName()));
+            $this->resetter->switchIndexAlias($indexName, true);
```
[the remark this answers is not in the document]
Comment (cableman) [reply src/Service/PopulateService.php:150]: Fixed both

File: src/Service/PopulateService.php (hunk 1 above)
Comment (cableman) [reply src/Service/PopulateService.php:66]: Lock added

Comment (turegjorup) [reply src/Service/PopulateService.php:66]: But in the Command, not the Service. What if the Service is use some place else...

**Review by turegjorup — APPROVED** [review]

**Review by turegjorup — APPROVED** [review]

**Pull request merged**